opendebates/moderator_views.py

```diff
@@ -1,42 +1,56 @@
 from djangohelpers.lib import rendered_with, allow_http
-from django.http import HttpResponse
-from django.shortcuts import redirect
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required
+from django.db.models import Count
+from django.http import HttpResponseNotFound
+from django.shortcuts import get_object_or_404, redirect
+from django.utils.translation import ugettext as _
 
 from opendebates_emails.models import send_email
-from .models import Submission, Vote
+from .models import Submission, Vote, Flag
 
 
 @rendered_with("opendebates/moderation/mark_duplicate.html")
 @allow_http("GET", "POST")
+@login_required
 def mark_duplicate(request):
     if not request.user.is_superuser:
-        return HttpResponse("forbidden")  # @@TODO
+        return HttpResponseNotFound()
 
     to_remove_default = ''
     if request.method == "GET":
         return {'to_remove_default': request.GET.get("to_remove", "")}
 
-    to_remove = Submission.objects.select_related("voter", "voter__user").get(
-        id=request.POST['to_remove'])
+    to_remove = get_object_or_404(Submission, pk=request.POST['to_remove'])
     try:
-        duplicate_of = Submission.objects.select_related(
-            "voter", "voter__user").get(id=request.POST['duplicate_of'])
+        duplicate_of = Submission.objects.get(id=request.POST['duplicate_of'])
```

Comment:
Should we add `approved=True` to this queryset and the one above (get_object_or_404)?